### PR TITLE
Validation scripts for all opcodes

### LINKED ISSUE #116: validate marginal/arguments for all OPCODEs

### PR DESCRIPTION
**Pull request by pdobacz**
closes #116 

(except RETURNDATACOPY)

### CODE REVIEW

File: src/analysis/validation_with_arguments.Rmd [hunk 1]
```diff
@@ -267,31 +295,38 @@ We divide the validation data into classes with respect to the length of the pro
 ```{r}
 validation_long = validation[which(validation$program_length > 400), ]
 validation_short = validation[which(validation$program_length <= 100), ]
+validation_no_exp = validation[which(validation$dominant != 'EXP'), ]
```
Comment (lukasz-glen) [inline src/analysis/validation_with_arguments.Rmd:298]: Why `no_exp`?

**Review by lukasz-glen — APPROVED** [review]
Just two questions

File: src/program_generator/pg_validation.py [hunk 2]
```diff
@@ -106,75 +176,23 @@ def _generate_random_arithmetic(self, opsLimit, bytecodeLimit, dominant, pushMax
     bytecode = ''
     # always preallocate memory to avoid uneven amount of allocation later
     bytecode += initial_mstore_bytecode()
+    # always include at least one JUMP
+    bytecode += jump_opcode_combo(bytecode, "56")
```
[the remark this answers is not in the document]
Comment (pdobacz) [reply src/program_generator/pg_validation.py:180]: there seems to be some constant preprocessing executed, when there is at least one JUMP (or JUMPDEST) present. If we don't include it globally, the validation programs containing at least one JUMP become de-facto outliers, despite the JUMP operation being relatively cheap marginally.

We compute marginal cost, and can safely assume that there will be at least one JUMP instruction in the code (e.g. the JUMP to a contract method).

File: src/analysis/validation_with_arguments.Rmd (hunk 1 above)
Comment (pdobacz) [reply src/analysis/validation_with_arguments.Rmd:298]: just for comparison, exp is a very expensive program and including it "fixes" the model in favor of the trivial model. We would like to ensure that even without exp we still improve fit compared to the trivial model (true).

**Pull request merged**